sumpy/expansion/__init__.py

```diff
@@ -244,31 +267,80 @@ def _spmv(spmat, x, sparse_vectors):
 # }}}
 
 
-class LinearRecurrenceBasedDerivativeWrangler(DerivativeWrangler):
+def _fast_spmv(reconstruct_matrix, vec, sac, transpose=False):
+    if not transpose:
+        res = [0] * len(reconstruct_matrix)
+        stored_idx = 0
+        for row, deps in enumerate(reconstruct_matrix):
+            if len(deps) == 0:
+                res[row] = vec[stored_idx]
+                stored_idx += 1
+            else:
+                for k, v in deps:
+                    res[row] += res[k] * v
+            new_sym = sym.Symbol(sac.assign_unique("expr", res[row]))
+            res[row] = new_sym
+        return res
+    else:
+        res = []
+        expr_all = list(vec)
+        for row, deps in reversed(list(enumerate(reconstruct_matrix))):
+            if len(deps) == 0:
+                res.append(expr_all[row])
+                continue
+            new_sym = sym.Symbol(sac.assign_unique("expr", expr_all[row]))
+            for k, v in deps:
+                expr_all[k] += new_sym * v
+        res.reverse()
+        return res
```

Comment:
Have a test for these somewhere? (Not 100% needed, since it's implicitly tested by the surrounding code... but it'd still be nice.)

Reply:
Done in ba952e87821c666ca47fa59b6ed649d405196763